pkg/shuffle/shuffle.go

```diff
@@ -0,0 +1,43 @@
+package shuffle
+
+import (
+	"github.com/numaproj/numaflow/pkg/isb"
+	"hash"
+	"hash/fnv"
+)
+
+// Shuffle shuffles messages among ISB
+type Shuffle struct {
+	bufferIdentifiers []string
+	hash              hash.Hash64
+}
+
+// NewShuffle accepts list of buffer identifiers(unique identifier of isb)
+// and returns new shuffle instance
+func NewShuffle(bufferIdentifiers []string) *Shuffle {
+	return &Shuffle{
+		bufferIdentifiers: bufferIdentifiers,
+		hash:              fnv.New64(),
+	}
+}
+
+// ShuffleMessages accepts list of isb messages and returns the mapping of isb to messages
+func (s *Shuffle) ShuffleMessages(messages []*isb.Message) map[string][]*isb.Message {
+	buffersCount := uint64(len(s.bufferIdentifiers))
```

Comment:
can't we store that into the `Shuffle` struct during `NewXXX`, so you don't have to call `len` for every `ShuffleMessages` call?

Reply:
true, changed